### PR TITLE
Implemented processing in case of GenerationType.IDENTITY and GenerationType.SEQUENCE by insert() method.

### PR DESCRIPTION
**Pull request by HidekiSugimoto189**
Implemented processing in case of GenerationType.IDENTITY by insert() method.

- TODO
  - Implementation of GenerationType.SEQUENCE.
  - Implementing ID numbering process with inserts() method.

Please review. @shout-star @ota-meshi

### CODE REVIEW

**Comment by HidekiSugimoto189** [comment]
Implemented processing in case of GenerationType.SEQUENCE.

**Comment by HidekiSugimoto189** [comment]
Implemented bulkInsert and batchInsert with IDENTITY or SEQUENCE.
and add many testcase.

File: src/main/java/jp/co/future/uroborosql/SqlAgentImpl.java [hunk 1]
```diff
@@ -271,7 +276,23 @@ public int update(final SqlContext sqlContext) throws SQLException {
 					if (maxRetryCount > 0 && getSqlConfig().getDialect().isRollbackToSavepointBeforeRetry()) {
 						setSavepoint(RETRY_SAVEPOINT_NAME);
 					}
-					return getSqlFilterManager().doUpdate(sqlContext, stmt, stmt.executeUpdate());
+					int count = getSqlFilterManager().doUpdate(sqlContext, stmt, stmt.executeUpdate());
+					if ((SqlKind.INSERT.equals(sqlContext.getSqlKind()) ||
+							SqlKind.BULK_INSERT.equals(sqlContext.getSqlKind()))
+							&& sqlContext.getGeneratedKeyColumns() != null
+							&& sqlContext.getGeneratedKeyColumns().length > 0) {
```
[the remark this answers is not in the document]
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlAgentImpl.java]: OK. Add SqlContext#hasGeneratedKeyColumns() method.

File: src/main/java/jp/co/future/uroborosql/SqlAgentImpl.java [hunk 2]
```diff
@@ -663,10 +699,53 @@ public int insert(final Object entity) {
...
+		if (int.class.equals(rawType) || Integer.class.equals(rawType)) {
+			column.setValue(entity, id.intValue());
+		} else if (long.class.equals(rawType) || Long.class.equals(rawType)) {
+			column.setValue(entity, id.longValue());
+		} else if (BigInteger.class.equals(rawType)) {
+			column.setValue(entity, id.toBigInteger());
+		} else if (BigDecimal.class.equals(rawType)) {
+			column.setValue(entity, id);
+		} else if (String.class.equals(rawType)) {
+			column.setValue(entity, id.toPlainString());
```
[the remark this answers is not in the document]
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlAgentImpl.java]: Add else clause and throws UroborosqlRuntimeException like other processing.

**Comment by shout-star** [comment]
エンジニアを褒めるネコ:おつかれさま

**Comment by shout-star** [comment]
refs #151

**Pull request merged**